set.mm

```diff
@@ -735799,6 +735836,340 @@ their group (addition) operations are equal for all pairs of elements of
+    cycsubmel $p |- ( X e. C <-> E. i e. NN0 X = ( i .x. A ) ) $=
+      ( wcel cv wceq cn0 wrex co ovex crn cfv eleq2i wfn wb fvelrnb ax-mp oveq1
+      fnmpti fvmpt eqeq1d eqcom syl6bb rexbiia 3bitri ) IDNIGUAZNZFOZGUBZIPZFQR
+      ZIURBESZPZFQRDUPIMUCGQUDUQVAUEAQAOZBESZGVDBETLUIFQIGUFUGUTVCFQURQNZUTVBIP
+      VCVFUSVBIAURVEVBQGVDURBEUHLURBETUJUKVBIULUMUNUO $.
+
+    $d A i $.  $d B i $.  $d .x. i $.
+    $( The set of nonnegative integer powers of an element ` A ` of a monoid
+       contains ` A ` .  (Contributed by AV, 28-Dec-2023.) $)
+    cycsubmcl $p |- ( A e. B -> A e. C ) $=
+      ( vi wcel cv co wceq cn0 wrex c1 1nn0 wb oveq1 eqeq2d adantl mulg1 eqcomd
+      a1i rspcedvd cycsubmel sylibr ) BCMZBLNZBEOZPZLQRBDMUKUNBSBEOZPZLSQSQMUKT
+      UGULSPZUNUPUAUKUQUMUOBULSBEUBUCUDUKUOBCEGBHIUEUFUHABCDELFGBHIJKUIUJ $.
+
+    $d A a b j k $.  $d B a b j k x $.  $d C a b $.  $d F j k $.
+    $d G a b i j k x $.  $d .x. j k $.
+    $( The set of nonnegative integer powers of an element ` A ` of a monoid
+       forms a submonoid containing ` A ` ( see ~ cycsubmcl ), called the
+       cyclic monoid generated by the element ` A ` .  This corresponds to the
+       statement in [Lang] p. 6.  (Contributed by AV, 28-Dec-2023.) $)
+    cycsubm $p |- ( ( G e. Mnd /\ A e. B ) -> C e. ( SubMnd ` G ) ) $=
+      ( va vi wcel wa cv co cn0 wceq cc0 vb vj vk cmnd csubmnd cfv wss c0g wral
+      cplusg crn mulgnn0cl 3expa an32s fmptd frnd eqsstrid wrex 0nn0 a1i eqeq2d
+      wb oveq1 adantl mulg0 eqcomd rspcedvd cycsubmel sylibr caddc simplr simpr
+      eqid wi nn0addcld adantr oveq12 ancoms simplll simpllr syl13anc sylan9eqr
+      mulgnn0dir exp32 rexlimdva com23 impd anbi12i 3imtr4g ralrimivv mpbir3and
+      w3a issubm ) GUDNZBCNZOZDGUEUFNZDCUGZGUHUFZDNZLPZUAPZGUJUFZQZDNZUADUILDUI
+      ZWPDFUKCKWPRCFWPARAPZBEQZCFWNXGRNZWOXHCNZWNXIWOXJCEGXGBHIULUMUNJUOUPUQWPW
+      SMPZBEQZSZMRURWTWPXMWSTBEQZSZMTRTRNWPUSUTXKTSZXMXOVBWPXPXLXNWSXKTBEVCVAVD
+      WPXNWSWOXNWSSWNCEGBWSHWSVMZIVEVDVFVGABCDEMFGWSHIJKVHVIWPXELUADDWPXAXLSZMR
+      URZXBUBPZBEQZSZUBRURZOXDUCPZBEQZSZUCRURZXADNZXBDNZOXEWPXSYCYGWPXRYCYGVNMR
+      WPXKRNZOZYCXRYGYKYBXRYGVNUBRYKXTRNZOZYBXRYGYMYBXROZOZYFXDXKXTVJQZBEQZSZUC
+      YPRYMYPRNYNYMXKXTWPYJYLVKZYKYLVLZVOVPYDYPSZYFYRVBYOUUAYEYQXDYDYPBEVCVAVDY
+      NYMXDXLYAXCQZYQXRYBXDUUBSXAXLXBYAXCVQVRYMYQUUBYMWNYJYLWOYQUUBSWNWOYJYLVSY
+      SYTWNWOYJYLVTCXCEGXKXTBHIXCVMZWCWAVFWBVGWDWEWFWEWGYHXSYIYCABCDEMFGXAHIJKV
+      HABCDEUBFGXBHIJKVHWHABCDEUCFGXDHIJKVHWIWJWNWQWRWTXFWLVBWOLUACXCDGWSHXQUUC
+      WMVPWK $.
+  $}
+
+
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Direct products (extension)
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  ${
+    $d G x y $.  $d U x y $.
+    smndlsmidm.p $e |- .(+) = ( LSSum ` G ) $.
+    $( The direct product is idempotent for monoids.  (Contributed by AV,
+       27-Dec-2023.) $)
```

Comment:
Also answering your question in the PR: ` ( LSSum ` G ) ` is simply the group sum (or magma law) considered on subsets (in other words, it is the associated "direct image" operation). So I would simply say "the group sum (magma law) as an operation on subsets".

The present statement is not an idempotence property of that operation. I'd state it as ` A .(+) U = U ` when ` 0g e. A C_ U ` and I'd say "A submonoid absorbs any of its subsets containing the identity under the monoid law operation on subsets."

Reply:
~smndlsmidm is (and was intended to be) a generalization of the already existing ~lsmidm. I think for the given restrictions the operation is idempotent. You are suggesting a broader generalization which could be proven in a separate theorem.

Reply:
Oh, I understand what you meant: "A submonoid/subgroup is idempotent for the group-like operation on subsets." (using my proposed terminology).

More generally: the equation $p + p = p$ expresses that " $p$ is idempotent for $+$".  But when you say "that operation is idempotent", it means "that operation applied twice is like applied once", or "a second application of that operation has no new effect", as in $f(f(x)) = f(x)$ for any $x$.

The two notions are related: adding an idempotent on the right is an idempotent operation. Indeed, if you define the function $f \colon G \to G, x \mapsto x + p$ where $p$ is idempotent for $+$, then $f$ is idempotent, that is, $f^2 = f$.

Reply:
According to Wikipedia [Idempotence](https://en.wikipedia.org/wiki/Idempotence):

> The binary operation ⋅ is said to be idempotent if x ⋅ x = x for all x ∈ S.


So we can say "The direct product is idempotent for submonoids". Iwill adjust the comments accordingly.

Reply:
Yes, since a binary operation is called idempotent when all its elements are idempotent, you can express "submonoids are idempotent for +" as "+ restricted to submonoids is idempotent". It's not false, but sounds a bit contrived. Your mathbox, so as you prefer.